wsd/COOLWSD.hpp

```diff
@@ -174,6 +174,30 @@ class WSProcess
         return false;
     }
 
+    bool sendUrpMessage(const std::string& message)
+    {
+        if (!_socket)
+            return false;
+        int urpFD = _socket->getIncomingFD(URPToKit);
+        if (urpFD == -1)
+            return false;
+        ssize_t len = message.size() - 4;
+        const char* buffer = message.data() + 4;
+        for (ssize_t i = 0; i < len;)
+        {
+            int written;
+            while ((written = ::write(urpFD, buffer + i, len - i)) < 0 && errno == EINTR)
+                LOG_TRC("EINTR writing to URP pipe: " << urpFD);
+            if (written < 0)
+            {
+                LOG_ERR("failed to write to URP pipe: " << urpFD << " : " << std::strerror(errno));
+                return false;
```

Comment:
This seems odd - can't we use a Stream to write from the coolwsd side and shuffle the data across ?